### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by rlapz**

### CODE REVIEW

File: rLapz/server.c [hunk 1]
```diff
@@ -1,7 +1,198 @@
...
+	printf(" -> File name length\t: %d\n", packet_data->filename_len);
+	printf(" -> File size\t\t: %zu byte\n", packet_data->file_size);
+	printf(" -> Stored at\t\t: %s\n", target_file);
+
+	/* reallocation memory size */
+	/* I know this kinda not safe */
+	data_arena = realloc(data_arena, sizeof(packet) + packet_data->file_size);
+	memset(data_arena, 0, sizeof(data_arena));
+	packet_data = (packet*)data_arena;
+
```
Comment (ammarfaizi2) [inline rLapz/server.c]: What does it do?

```c
memset(data_arena, 0, sizeof(data_arena)); // Why sizeof(data_arena)?
```

~~You probably want to write~~
~~memset(data_arena, 0, sizeof(packet) + packet_data->file_size);~~

**Comment by ammarfaizi2** [comment]
I compiled your code with clang and got a lot of warnings. And I think you really want to address some.

```
clang -Wall -Wextra -Weverything client.c -o client
client.c:45:31: warning: implicit conversion loses integer precision: 'int' to 'uint16_t' (aka 'unsigned short') [-Wimplicit-int-conversion]
        return inet_handler(argv[1], atoi(argv[2]), argv[3]);
               ~~~~~~~~~~~~          ^~~~~~~~~~~~~
client.c:38:36: warning: unused parameter 'envp' [-Wunused-parameter]
main(int argc, char *argv[], char *envp[])
                                   ^
client.c:89:30: warning: implicit conversion loses integer precision: 'unsigned long' to 'uint8_t' (aka 'unsigned char') [-Wimplicit-int-conversion]
        packet_data->filename_len = strlen(basename);
                                  ~ ^~~~~~~~~~~~~~~~
client.c:90:34: warning: implicit conversion changes signedness: '__off_t' (aka 'long') to 'uint64_t' (aka 'unsigned long') [-Wsign-conversion]
        packet_data->file_size = s_file.st_size;
                               ~ ~~~~~~~^~~~~~~
client.c:150:54: warning: implicit conversion changes signedness: 'ssize_t' (aka 'long') to 'size_t' (aka 'unsigned long') [-Wsign-conversion]
                send_bytes = send(client_fd, packet_data->content, read_bytes, 0);
                             ~~~~                                  ^~~~~~~~~~
client.c:151:18: warning: result of comparison of unsigned expression < 0 is always false [-Wtautological-unsigned-zero-compare]
                if (sent_bytes < 0) {
                    ~~~~~~~~~~ ^ ~
client.c:156:17: warning: implicit conversion changes signedness: 'ssize_t' (aka 'long') to 'unsigned long' [-Wsign-conversion]
                sent_bytes += read_bytes;
                           ~~ ^~~~~~~~~~
7 warnings generated.
clang -Wall -Wextra -Weverything server.c -o server
server.c:43:31: warning: implicit conversion loses integer precision: 'int' to 'uint16_t' (aka 'unsigned short') [-Wimplicit-int-conversion]
        return inet_handler(argv[1], atoi(argv[2]));
               ~~~~~~~~~~~~          ^~~~~~~~~~~~~
server.c:36:36: warning: unused parameter 'envp' [-Wunused-parameter]
main(int argc, char *argv[], char *envp[])
                                   ^
server.c:177:56: warning: implicit conversion changes signedness: 'ssize_t' (aka 'long') to 'size_t' (aka 'unsigned long') [-Wsign-conversion]
                write_bytes = write(file_desc, packet_data->content, recv_bytes);
                              ~~~~~                                  ^~~~~~~~~~
server.c:183:18: warning: implicit conversion changes signedness: 'ssize_t' (aka 'long') to 'unsigned long' [-Wsign-conversion]
                total_bytes += recv_bytes;
                            ~~ ^~~~~~~~~~
server.c:120:35: warning: unused parameter 'server_fd' [-Wunused-parameter]
file_handler(int *client_fd, int *server_fd)
                                  ^
server.c:135:23: warning: comparison of integers of different signs: 'ssize_t' (aka 'long') and 'unsigned long' [-Wsign-compare]
        while (recv_ok_bytes < sizeof(packet)) {
               ~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~
server.c:158:31: warning: 'memset' call operates on objects of type 'char' while the size is based on a different type 'char *' [-Wsizeof-pointer-memaccess]
        memset(data_arena, 0, sizeof(data_arena));
               ~~~~~~~~~~            ^~~~~~~~~~
server.c:158:31: note: did you mean to provide an explicit length?
        memset(data_arena, 0, sizeof(data_arena));
                                     ^~~~~~~~~~
7 warnings generated.
```

**Comment by ammarfaizi2** [comment]
Anyway, I didn't write my example well too. So this is probably acceptable (for simple file transfer).

File: rLapz/server.c (hunk 1 above)
Comment (ammarfaizi2) [reply rLapz/server.c]: Oh no!
You probably want to get rid of `memset`.

Why did you `memset` it anyway?
Doesn't it clobber the `filename_len` and `filename`?

Comment (rlapz) [reply rLapz/server.c]: I want to "clear" whole data inside `data_arena` including `filename_len` and `filename`, but I seems wrong.

**Comment by rlapz** [comment]
woah,
I want to fix it later. :D

File: rLapz/server.c (hunk 1 above)
Comment (rlapz) [reply rLapz/server.c]: wait, I forget to "backup" `filename_len`, `file_size`, and `filename` data to new variable,

**Review by ammarfaizi2 — APPROVED** [review]
Looks good to me

**Comment by ammarfaizi2** [comment]
Good job

**Pull request merged**